sklearn/experimental/tests/test_column_transformer.py

```diff
@@ -0,0 +1,279 @@
+
+    ct = ColumnTransformer([('trans', Trans(), [0, 1])])
+    assert_array_equal(ct.fit_transform(X_df), X_res_both)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_both)
+
+    ct = ColumnTransformer([('trans1', Trans(), [0]),
+                            ('trans2', Trans(), [1])])
+    assert_array_equal(ct.fit_transform(X_df), X_res_both)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_both)
+
+    # slice
+    ct = ColumnTransformer([('trans', Trans(), slice(0, 1))])
+    assert_array_equal(ct.fit_transform(X_df), X_res_first2D)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_first2D)
+
+    ct = ColumnTransformer([('trans', Trans(), slice(0, 2))])
+    assert_array_equal(ct.fit_transform(X_df), X_res_both)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_both)
+
+    # boolean mask
+    ct = ColumnTransformer([('trans', Trans(), np.array([True, False]))])
+    assert_array_equal(ct.fit_transform(X_df), X_res_first2D)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_first2D)
+
+    s_mask = pd.Series([True, False], index=['first', 'second'])
+    ct = ColumnTransformer([('trans', Trans(), s_mask)])
+    assert_array_equal(ct.fit_transform(X_df), X_res_first2D)
+    assert_array_equal(ct.fit(X_df).transform(X_df), X_res_first2D)
+
+    # test with transformer_weights
+    transformer_weights = {'trans1': .1, 'trans2': 10}
+    both = ColumnTransformer([('trans1', Trans(), ['first']),
+                              ('trans2', Trans(), ['second'])],
+                             transformer_weights=transformer_weights)
+    res = np.vstack([transformer_weights['trans1'] * X_df['first'],
+                     transformer_weights['trans2'] * X_df['second']]).T
+    assert_array_equal(both.fit_transform(X_df), res)
+    assert_array_equal(both.fit(X_df).transform(X_df), res)
+
+    # test multiple columns
+    both = ColumnTransformer([('trans', Trans(), ['first', 'second'])],
+                             transformer_weights={'trans': .1})
+    assert_array_equal(both.fit_transform(X_df), 0.1 * X_res_both)
+    assert_array_equal(both.fit(X_df).transform(X_df), 0.1 * X_res_both)
+
+    both = ColumnTransformer([('trans', Trans(), [0, 1])],
+                             transformer_weights={'trans': .1})
+    assert_array_equal(both.fit_transform(X_df), 0.1 * X_res_both)
+    assert_array_equal(both.fit(X_df).transform(X_df), 0.1 * X_res_both)
+
+    # ensure pandas object is passes through
+
+    class TransAssert(BaseEstimator):
+
+        def fit(self, X, y=None):
+            return self
+
+        def transform(self, X, y=None):
+            assert_true(isinstance(X, (pd.DataFrame, pd.Series)))
+
+    ct = ColumnTransformer([('trans', TransAssert(), 'first')])
+    ct.fit_transform(X_df)
+    ct = ColumnTransformer([('trans', TransAssert(), ['first', 'second'])])
+    ct.fit_transform(X_df)
+
+
+def test_column_transformer_sparse_array():
```

Comment:
can we maybe do these as loops over X or write a function? this seems like a lot of duplication.